src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -868,3 +868,248 @@ def describe_non_ready_agent_log(filepath_with_timestamp, corev1_api_instance, a
+
+    try:
+        if storage_space_available:
+            command = [helm_client_location, "get", "values", "azure-arc", "-n", release_namespace, "--output", "json"]
+            if kube_config:
+                command.extend(["--kubeconfig", kube_config])
+            if kube_context:
+                command.extend(["--kube-context", kube_context])
+            # Using Popen to execute the command and fetching the output
+            response_kubectl_get_events = Popen(command, stdout=PIPE, stderr=PIPE)
+            output_kubectl_get_events, error_kubectl_get_events = response_kubectl_get_events.communicate()
+            if response_kubectl_get_events.returncode != 0:
+                telemetry.set_exception(exception=error_kubectl_get_events.decode("ascii"), fault_type=consts.Helm_Get_Values_Failed_Fault_Type, summary='Error while doing helm get values for azure-arc release')
+                logger.warning("Error while doing helm get values for azure-arc release. We were not able to capture this log in arc_diganostic_logs folder. Exception: ", error_kubectl_get_events.decode("ascii"))
+                diagnoser_output.append("Error while doing helm get values for azure-arc release. We were not able to capture this log in arc_diganostic_logs folder. Exception: " + error_kubectl_get_events.decode("ascii"))
+                return storage_space_available
+
+            # Converting output obtained in json format
+            events_json = output_kubectl_get_events.decode("ascii")
+            events_json = json.loads(events_json)
+            try:
+                if(events_json['global']['onboardingPrivateKey']):
+                    del events_json['global']['onboardingPrivateKey']
+            except Exception as e:
+                pass
+
+            events_json = yaml.dump(events_json)
+            # Path to add the helm values of azure-arc release
+            event_logs_path = os.path.join(filepath_with_timestamp, "helm_values_azure_arc.txt")
+
+            with open(event_logs_path, 'w') as event_log:
+                event_log.write(events_json)
+
+            return storage_space_available
+
+    # For handling storage or OS exception that may occur during the execution
+    except OSError as e:
+        if "[Errno 28]" in str(e):
+            storage_space_available = False
+            telemetry.set_exception(exception=e, fault_type=consts.No_Storage_Space_Available_Fault_Type, summary="No space left on device")
+            shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+        else:
+            logger.warning("An exception has occured while storing helm values of azure-arc release in the user local machine. Exception: {}".format(str(e)) + "\n")
+            telemetry.set_exception(exception=e, fault_type=consts.Helm_Get_Values_Failed_Fault_Type_Exception, summary="Exception occured while storing helm values of azure-arc release")
```

Comment:
here fault type should be something like "helm_values_save_failed" right? the current fault type is misleading. It is giving the meaning that helm get values failed

Reply:
similar comment for other 3-4 fault types where storage failed